core/control-plane/transfer-core/src/main/java/org/eclipse/edc/connector/transfer/edr/EndpointDataReferenceReceiverRegistryImpl.java

```diff
@@ -40,14 +41,18 @@ public void registerReceiver(@NotNull EndpointDataReferenceReceiver receiver) {
 
     @Override
     public @NotNull CompletableFuture<Result<Void>> receiveAll(@NotNull EndpointDataReference edr) {
-        return receivers.stream()
-                .map(receiver -> receiver.send(edr))
-                .collect(asyncAllOf())
-                .thenApply(results -> results.stream()
-                        .filter(Result::failed)
-                        .findFirst()
-                        .map(failed -> Result.<Void>failure(failed.getFailureMessages()))
-                        .orElse(Result.success()))
-                .exceptionally(throwable -> Result.failure("Failed to receive endpoint data reference: " + throwable.getMessage()));
+        if (!receivers.isEmpty()) {
```

Comment:
Better flip the `if` condition, when a branch is one liner is always better to have it visible at the top